cmd/limactl/disk.go

```diff
@@ -418,3 +421,62 @@ func diskResizeAction(cmd *cobra.Command, args []string) error {
 func diskBashComplete(cmd *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
 	return bashCompleteDiskNames(cmd)
 }
+
+// func newDiskRegisterCommand()
+func newDiskAddCommand() *cobra.Command {
+	diskAddCommand := &cobra.Command{
+		Use: "add existing DISK to Lima",
+		Example: `
+Add a disk:
+$ limactl disk add DISK`,
+		Short:             "Add existing to Lima",
+		Args:              WrapArgsError(cobra.ExactArgs(1)),
+		RunE:              diskAddAction,
+		ValidArgsFunction: diskBashComplete,
+	}
+
+	diskAddCommand.Flags().String("filename", "", "Path to disk image")
+	_ = diskAddCommand.MarkFlagRequired("filename")
+	return diskAddCommand
+}
+
+func diskAddAction(cmd *cobra.Command, args []string) error {
+	fName, err := cmd.Flags().GetString("filename")
+	if err != nil {
+		return err
+	}
+
+	f, err := os.Open(fName)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	img, err := qcow2reader.Open(f)
+	if err != nil {
+		return err
+	}
+
+	diskName := args[0]
+	diskSize := img.Size()
+	format := img.Type()
+
+	switch format {
+	case "qcow2", "raw":
+	default:
+		return fmt.Errorf(`disk format %q not supported, use "qcow2" or "raw" instead`, format)
+	}
+
+	diskDir, err := store.DiskDir(diskName)
+	if err := os.MkdirAll(diskDir, 0o755); err != nil {
+		return err
+	}
+
+	if err := os.Symlink(fName, filepath.Join(diskDir, "datadisk")); err != nil {
```

Comment:
This should use https://pkg.go.dev/github.com/containerd/continuity@v0.4.5/fs#CopyFile

`CopyFile` attempts CoW to save the disk space and the time when it is available.